settings.py

```diff
@@ -111,49 +112,32 @@ def register_components():
 
         hubs_context['registered_hubs_components'][component_name] = component_class
 
-@persistent
-def load_handler(_dummy):
-    reload_context(bpy.context.scene.hubs_settings.config_path)
 
-def config_updated(self, _context):
-    load_handler(self)
+def config_updated(_self, context):
+    reload_context(context)
+
+class HubsAddonPreferences(AddonPreferences):
+    bl_idname = __package__
 
-class HubsSettings(PropertyGroup):
     config_path: StringProperty(
-        name="config_path",
-        description="Path to the config file",
+        name="Component Config",
         default=default_config_path,
-        options={'HIDDEN'},
-        maxlen=1024,
         subtype='FILE_PATH',
         update=config_updated
     )
 
-    @property
-    def hubs_config(self):
-        global hubs_context
-        return hubs_context['hubs_config']
-
-    @property
-    def registered_hubs_components(self):
-        return hubs_context['registered_hubs_components']
-
-    @property
-    def registered_hubs_classes(self):
-        return hubs_context['registered_hubs_classes']
-
-    def reload_config(self):
-        reload_context(self.config_path)
+    def draw(self, context):
+        layout = self.layout.row()
+        layout.prop(self, "config_path")
+        layout.operator(ReloadHubsConfig.bl_idname, text="", icon="FILE_REFRESH")
 
 def register():
-    bpy.utils.register_class(HubsSettings)
```

Comment:
I'm a little worried about how this will impact existing component data, but I don't have a good project to test with. It could be mostly fine. You may just need to update the path to the Hubs components config.